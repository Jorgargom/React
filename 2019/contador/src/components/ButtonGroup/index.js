import React from 'react';
import PropTypes from 'prop-types';
import Flex from '../Flex';

const ButtonGroup = ({ center, ...props }) => (
  <Flex justifyContent={center ? 'center' : 'space-between'} {...props} />
);

ButtonGroup.defaultProps = {
  center: false,
};

ButtonGroup.propTypes = {
  center: PropTypes.bool,
};

export default ButtonGroup;
