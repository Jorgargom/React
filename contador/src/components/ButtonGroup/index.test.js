import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ButtonGroup from '.';

describe('ButtonGroup', () => {
  const CHILD = 'FOO';
  const component = <ButtonGroup>{CHILD}</ButtonGroup>;
  test('it shallows', () => {
    const wrapper = shallow(component);
    expect(wrapper.text()).toBe(CHILD);
    expect(wrapper).toBeDefined();
  });
  test('snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
