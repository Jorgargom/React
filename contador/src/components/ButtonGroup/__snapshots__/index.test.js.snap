// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<div
  className="sc-bdVaJa eMyZqx"
>
  FOO
</div>
`;

exports[`ButtonGroup snapshot 1`] = `
<div
  className="sc-bdVaJa eMyZqx"
>
  FOO
</div>
`;
